Brridge-Group/Golden-Gate-Hairr
# Feature: Dynamically render salon Services and Features from the database to the Business Profile creation form

## Description

This Pull Request (PR) adds a new feature that allows for `GET` requests to the database to fetch all the Services and Features and render all items to the `<BusinessProfile/>` component in the business profile creation form. 

## Related Issue/Ticket

Closes #34

## Acceptance Criteria

- [x] In the business profile form, show the Features dynamically that exist in the database
- [x] In the business profile form, show the Services dynamically that exist in the database

## Type of Changes

|     |  Type                       |
| --- | -------------------------- |
|     | :bug: Bug fix              |
| ✓ | :sparkles: New feature     |
| ✓ | :hammer: Refactoring       |
|     | :100: Add tests            |
|     | :link: Update dependencies |
|     | :scroll: Docs              |

## Updates

### Before
![image](https://user-images.githubusercontent.com/32943879/137066096-1761bd5f-b6cd-4b42-8a24-cdbfd674ead1.png)
![image](https://user-images.githubusercontent.com/32943879/143140272-f0b3ae1d-646e-4ae1-90f2-3002f32c4557.png)

### After
![image](https://user-images.githubusercontent.com/32943879/143147754-0cb11f92-ca9a-425e-a681-6b71498f5bf1.png)
![image](https://user-images.githubusercontent.com/32943879/143148429-9218e6fc-36a5-4267-8330-19da4ca22fdf.png)
![image](https://user-images.githubusercontent.com/32943879/143148224-076a6ee8-60b2-4cd8-a134-76e7788e57e1.png)
![image](https://user-images.githubusercontent.com/32943879/143068234-e5b03db5-ed9b-4192-9c7a-ed8ca4307516.png)
![image](https://user-images.githubusercontent.com/32943879/143148056-be38742e-ac5c-4309-84b3-0e11bc31c1d3.png)

## Testing Steps / QA Criteria
 
- From your terminal, run git checkout `hair-34-dynamic-feat-services`.
- Once on the `hair-34-dynamic-feat-services` branch, run npm start in the root and client directories, respectively, to load the server.
- If one is an existing user, visit the `<BusinessProfile/>` component at http://localhost:3000/business/profile  to create a business profile.
- If one is not an existing user, visit http://localhost:3000/signup to sign up and ensure to select 'Check If True' to see the `<BusinessProfile/>` component at http://localhost:3000/business/profile to create a business profile.
- To test adding a new service or feature, in Postman or Insomnia, initiate a `POST` request to the appropriate URL based on the following format:
	- Services URL: localhost:3000/api/services

	- Features URL: localhost:3000/api/features

``` json
{
  "name": "", //required
  "description": "" // required
}
```
After successfully adding the new item, it should be viewable while logged in and visiting the `<BusinessProfile/>` component at http://localhost:3000/business/profile and in the remote MongoDB.
